source/base/timer.cc

```diff
@@ -353,8 +353,13 @@ TimerOutput::TimerOutput (MPI_Comm      mpi_communicator,
 
 TimerOutput::~TimerOutput()
 {
-  while (active_sections.size() > 0)
-    leave_subsection();
+  try
+    {
+      while (active_sections.size() > 0)
+        leave_subsection();
```

Comment:
Hm, I'm not terribly excited about wrapping content of destructors into try-catch clauses just because some function can throw :-( But then I also don't have a good solution otherwise...